scripts/lib/CIME/XML/archive.py

```diff
@@ -26,10 +26,9 @@ def setup(self, env_archive, components, files=None):
         if files is None:
             files = Files()
 
-        components_node = ET.Element("components")
+        components_node = ConstantElement(ET.Element("components"))
```

Comment:
I think that the general philosophy has been that xml should not be referenced directly outside of the XML directory.   I consider this a bug.

Reply:
Yes, this is definitely a leak of XML encapsulation.